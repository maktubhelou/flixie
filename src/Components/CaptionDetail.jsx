import React from 'react'
import PropTypes from 'prop-types'

const styles = {
  left: {
    marginBottom: '8px',
    padding: '8px',
    float: 'left',
    backgroundColor: '#111133',
  },
  right: {
    marginBottom: '8px',
    padding: '8px',
    float: 'right',
    backgroundColor: '#111133',
  },
}

const CaptionDetail = ({ title, content, left }) => (
  <div style={left ? styles.left : styles.right}>
    <strong>{`${title}: `}</strong>
    {content}
  </div>
)

CaptionDetail.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  left: PropTypes.bool,
}

CaptionDetail.defaultProps = {
  left: false,
}

export default CaptionDetail
