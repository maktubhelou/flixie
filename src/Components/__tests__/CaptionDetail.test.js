import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import CaptionDetail from '../CaptionDetail'

describe('<CaptionDetail />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<CaptionDetail />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('matches snapshot', () => {
    const tree = renderer
      .create(<CaptionDetail title="Rating" content="10.0" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
