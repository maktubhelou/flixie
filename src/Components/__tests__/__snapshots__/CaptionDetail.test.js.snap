// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaptionDetail /> matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#111133",
      "float": "right",
      "marginBottom": "8px",
      "padding": "8px",
    }
  }
>
  <strong>
    Rating: 
  </strong>
  10.0
</div>
`;
